client/client.go

```diff
@@ -22,12 +21,18 @@ func SetLogLevel(level string) error {
 // GetConnection attempts to return a secure connection to a gRPC server at a given endpoint.
 // Note that several clients can be passed the same connection object, as the gRPC framework
 // is able to multiplex several RPCs on the same connection, thus reducing the overhead
-func GetConnection(serverEndpoint, caCert string) (*grpc.ClientConn, error) {
+func GetConnection(serverEndpoint, caCert string, insecure bool) (*grpc.ClientConn, error) {
 	logger.Info("Getting the connection")
 	timeoutSec := config.LoadTimeout()
 
+	// Notify the end user about security implications when running in insecure mode
+	if insecure {
+		logger.Warning(`######## You requested an *insecure* channel! ######## As a consequence,
+			server's identity will *NOT* be validated! Please consider using a secure connection`)
```

Comment:
FYI: if you use backticks in Go, then it will take all the text *literally*, including  the three tabs at the beginning of your second line. Perhaps it would be nicer to have to separate warning lines, but I'm not too picky about this.